### DISCUSSION TITLE
Can invalidate trigger hooks.ts?

### DISCUSSION THREAD

**Discussion by xpluscal** [sveltejs/kit#7092] [Q&A]
I am currently using jwt-token based authentication.
The jwt is stored in a httpOnly cookie.
Now if the cookie expires, svelte returns /__invalid, which causes load operations to re-run but hooks.ts don't seem to.

The desired behaviour would be to have the page re-run the hook, so it detects that the session ran out and redirects the user to login page (that part is already in hooks, just doesn't get called on invalidate)

Appreciate any help.

**Comment by teemingc** [comment]
I've tested and was able to get `hooks` to run on [`invalidateAll()`](https://kit.svelte.dev/docs/modules#$app-navigation-invalidateall)

Here's a reproduction:
https://stackblitz.com/edit/sveltejs-kit-template-default-exvv2x?file=src/hooks.server.js

**Reply by yousufiqbal** [reply]
This is how I do it.
